warp10/src/main/java/io/warp10/script/functions/TOPICKLE.java

```diff
@@ -16,32 +16,54 @@
 
 package io.warp10.script.functions;
 
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GTSPickler;
+import io.warp10.continuum.gts.GeoTimeSerie;
 import io.warp10.script.NamedWarpScriptFunction;
 import io.warp10.script.WarpScriptException;
 import io.warp10.script.WarpScriptStack;
 import io.warp10.script.WarpScriptStackFunction;
+import io.warp10.script.WarpScriptLib;
+
 
 import java.io.IOException;
 
+import net.razorvine.pickle.Pickler;
 import net.razorvine.pyro.serializer.PickleSerializer;
 
 /**
  * Encode an object as Pickle
  */
 public class TOPICKLE extends NamedWarpScriptFunction implements WarpScriptStackFunction {
-  
+
+  // register pickler for custom classes
+  static {
+    Pickler.registerCustomPickler(GeoTimeSerie.class, new GTSPickler());
+  }
+
   public TOPICKLE(String name) {
     super(name);
   }
   
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
-    Object o = stack.pop();
+    Object o = stack.peek();
+
+    if (o instanceof GTSEncoder) {
+      ((TOGTS) WarpScriptLib.getFunction(WarpScriptLib.TOGTS)).apply(stack);
```

Comment:
You need to allocate your own instance in the static block, because this function may have been disabled or redefined by an extension, so retrieving it in WarpScriptLib is not safe